src/Mvc/Mvc.Core/src/DependencyInjection/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -86,6 +87,24 @@ private static ApplicationPartManager GetApplicationPartManager(IServiceCollecti
         }
 
         return manager;
+
+        static string? GetEntryAssemblyName(IWebHostEnvironment? environment)
+        {
+            if (environment is null)
+            {
+                return null;
+            }
+
+            try
+            {
+                _ = Assembly.Load(new AssemblyName(environment.ApplicationName));
```

Comment:
I'm not really happy with that solution.

What if we extend `IWebHostEnvironment` with a property for the `HostingStartupAssemblyName`?
Then these checks just need to be done once, the property-value can be used whereever needed.

Also the logic for determing the correct assembly name is only in one place.